### PR TITLE
feat: Support `*`-wildcard postfix in `refKeys` for `RelationalBones`

### PR DESCRIPTION
**Pull request by ArneGudermann**
proposal for #1019.

### CODE REVIEW

File: src/viur/core/skeleton.py [hunk 1]
```diff
@@ -1338,7 +1338,14 @@ def fromSkel(cls, kindName: str, *args: List[str]) -> Type[RefSkel]:
...
         fromSkel = skeletonByKind(kindName)
-        newClass.__boneMap__ = {k: v for k, v in fromSkel.__boneMap__.items() if k in args}
+        bone_map = {}
+        for arg in args:
+            if "*" in arg:  # we have a wild card
+                bone_map |= {k: v for k, v in fromSkel.__boneMap__.items() if k.startswith(arg.replace("*", ""))}
+            else:
+                if bone := fromSkel.__boneMap__.get(arg):
+                    bone_map[arg] = bone
+        newClass.__boneMap__ = bone_map
```
Comment (sveneberth) [inline src/viur/core/skeleton.py:1348]: That looks complicated. Couldn't we simplify this with [`utils.string.is_prefix()`](https://github.com/viur-framework/viur-core/blob/c34d45fdf0eeeace4d4317ab89cd251ddf6191cf/src/viur/core/utils/string.py#L88): 

```suggestion
        for arg in args:
            bone_map |= {k: v for k, v in fromSkel.__boneMap__.items()
                         if utils.string.is_prefix(k, arg, "*"))}
        newClass.__boneMap__ = bone_map
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

File: src/viur/core/skeleton.py (hunk 1 above)
Comment (ArneGudermann) [reply src/viur/core/skeleton.py:1348]: But now we iter over the bonemap `len(args) `time ? That is not so good i think.

Comment (sveneberth) [reply src/viur/core/skeleton.py:1348]: > But now we iter over the bonemap `len(args) `time ? That is not so good i think.

well, okay.


But I think index access would be faster than an _in_ test. And slicing the prefix once would be better than _replace_ on every _if_ check. But I didn't bench marked this ...


```suggestion
        for arg in args:
            if arg[-1] == "*":  # we have a wild card prefix
                prefix = arg[:-1]
                bone_map |= {k: v for k, v in fromSkel.__boneMap__.items() if k.startswith(prefix)}
            else:
                if bone := fromSkel.__boneMap__.get(arg):
                    bone_map[arg] = bone
        newClass.__boneMap__ = bone_map
```

**Review by sveneberth — CHANGES_REQUESTED** [review]
I think there are more changes necessary. We valide the `refKeys` also before filtering/ordering:

https://github.com/search?q=repo%3Aviur-framework%2Fviur-core+self.refKeys++path%3A**%2Fsrc%2Fviur%2Fcore%2Fbones%2Frelational.py&type=code

We should resolve the wildcard in the bone in `getSystemInitialized()`. Maybe this could already be enough:
```py
self.refKeys = list(self._refSkelCache__boneMap__.keys())
```

File: src/viur/core/skeleton.py [hunk 2]
```diff
@@ -1363,8 +1363,15 @@ def fromSkel(cls, kindName: str, *args: list[str]) -> t.Type[RefSkel]:
         """
         newClass = type("RefSkelFor" + kindName, (RefSkel,), {})
         fromSkel = skeletonByKind(kindName)
-        newClass.__boneMap__ = {k: v for k, v in fromSkel.__boneMap__.items() if k in args}
-        newClass.kindName = kindName
+        bone_map = {}
```
Comment (sveneberth) [inline src/viur/core/skeleton.py:1366]:
```suggestion        
        newClass.kindName = kindName
        bone_map = {}
```

**Review by sveneberth — CHANGES_REQUESTED** [review]

**Review by sveneberth — DISMISSED** [review]

**Comment by phorward** [comment]
@ArneGudermann what do you think, should we use fnmatch here as well? This would also allow to match `*_bones` next to `a*bones` ... I think this would be very nice!

**Comment by ArneGudermann** [comment]
@phorward Now better ?

File: src/viur/core/skeleton.py [hunk 3]
```diff
@@ -1363,8 +1364,15 @@ def fromSkel(cls, kindName: str, *args: list[str]) -> t.Type[RefSkel]:
...
         fromSkel = skeletonByKind(kindName)
-        newClass.__boneMap__ = {k: v for k, v in fromSkel.__boneMap__.items() if k in args}
         newClass.kindName = kindName
+        bone_map = {}
+        for arg in args:
+            if arg[-1] == "*":  # we have a wild card prefix
+                bone_map |= {k: v for k, v in fromSkel.__boneMap__.items() if fnmatch.fnmatch(k, arg)}
+            else:
+                if bone := fromSkel.__boneMap__.get(arg):
+                    bone_map[arg] = bone
```
Comment (phorward) [inline src/viur/core/skeleton.py:1374]: > @phorward Now better ?

Quite, but it's still inconsistent, and only accepts keys ending with "*". 
```suggestion
        for arg in args:
            bone_map |= {k: fromSkel.__boneMap__[k] for k in fnmatch.filter(fromSkel.__boneMap__.keys(), arg)}
```
What do you think about this?

**Review by phorward — CHANGES_REQUESTED** [review]

**Review by phorward — APPROVED** [review]
@sveneberth are you still fine with this?

**Review by sveneberth — APPROVED** [review]

**Pull request merged**